src/backend/distributed/commands/table.c

```diff
@@ -219,14 +219,22 @@ PostprocessCreateTableStmtPartitionOf(CreateStmt *createStatement, const
 	if (IsCitusTable(parentRelationId))
 	{
 		Oid relationId = RangeVarGetRelid(createStatement->relation, NoLock, missingOk);
-		Var *parentDistributionColumn = DistPartitionKeyOrError(parentRelationId);
-		char parentDistributionMethod = DISTRIBUTE_BY_HASH;
-		char *parentRelationName = generate_qualified_relation_name(parentRelationId);
-		bool viaDeprecatedAPI = false;
-
-		CreateDistributedTable(relationId, parentDistributionColumn,
-							   parentDistributionMethod, ShardCount,
-							   parentRelationName, viaDeprecatedAPI);
+
+		/*
+		 * If that partition is already distributed, or if the table is not a partition
+		 * of the current parent, just skip like in plain PG.
+		 */
+		if (!(IsCitusTable(relationId)) && PartitionTable(relationId) &&
+			(PartitionParentOid(relationId) == parentRelationId))
```

Comment:
minor: you could explain this `PartitionParentOid(relationId) == parentRelationId` check in the comment, since it's a little bit difficult to see when reading this code piece.

Reply:
It is already explained in the comment
`...or if the table is not a partition of the current parent...`